app/assets/javascripts/controllers/report_data_controller.js

```diff
@@ -0,0 +1,357 @@
+  * @param {Object} event jQuery event.
+  * @returns {undefined}
+  */
+  ReportDataController.prototype.onItemClicked = function(item, event) {
+    event.stopPropagation();
+    event.preventDefault();
+    var prefix = this.initObject.showUrl;
+    var splitUrl = this.initObject.showUrl.split('/');
+    if (this.initObject.isExplorer && isCurrentControllerOrPolicies(splitUrl)) {
+      var itemId = item.id;
+      if (this.initObject.showUrl.indexOf('?id=') !== -1 ) {
+        itemId = this.initObject.showUrl.indexOf('xx-') !== -1 ? '_-' + item.id : '-' + item.id;
+      }
+      var url = prefix + itemId;
+      $.post(url).always(function() {
+        this.setExtraClasses();
+      }.bind(this));
+    } else {
+      prefix = prefix[prefix.length - 1] !== '/' ? prefix + '/' : prefix;
+      this.$window.DoNav(prefix + item.id);
+    }
+    return false;
+  };
+
+  /**
+  * Method which will be fired when item was selected (either trough select box or by clicking on tile).
+  * @param {Object} item which item was selected.
+  * @param {Boolean} isSelected true | false.
+  * @returns {undefined}
+  */
+  ReportDataController.prototype.onItemSelect = function(item, isSelected) {
+    var selectedItem = _.find(this.gtlData.rows, {id: item.id});
+    if (selectedItem) {
+      selectedItem.checked = isSelected;
+      selectedItem.selected = isSelected;
+      this.$window.sendDataWithRx({rowSelect: selectedItem});
+      if (isSelected) {
+        ManageIQ.gridChecks.push(item.id);
+      } else {
+        var index = ManageIQ.gridChecks.indexOf(item.id);
+        index !== -1 && ManageIQ.gridChecks.splice(index, 1);
+      }
+    }
+  };
+
+  ReportDataController.prototype.initObjects = function(initObject) {
+    this.gtlData = { cols: [], rows: [] };
+    this.initObject = initObject;
+    if (this.initObject.showUrl === '') {
+      this.initObject.showUrl = '/' + ManageIQ.controller;
+      if (this.initObject.isExplorer) {
+        this.initObject.showUrl += '/x_show/';
+      } else {
+        this.initObject.showUrl += '/show/';
+      }
+    }
+    this.gtlType = initObject.gtlType || 'grid';
+    this.settings.isLoading = true;
+    ManageIQ.gridChecks = [];
+    this.$window.sendDataWithRx({setCount: 0});
+  };
+
+  /**
+  * Method for initializing controller. Good for bootstraping controller after loading items. This method will call
+  * getData for fetching data for current state. After these data were fetched, sorting items and filtering them takes
+  * place.
+  * @param {Object} initObject this object will hold all information about current state.
+  * ```
+  *   initObject: {
+  *     modelName: string,
+  *     gtlType: string,
+  *     activeTree: string,
+  *     currId: string,
+  *     isExplorer: Boolean
+  *   }
+  * ```
+  * @returns {Object} promise of fetched data.
+  */
+  ReportDataController.prototype.initController = function(initObject) {
+    this.$window.ManageIQ.gtl = this.$window.ManageIQ.gtl || {};
```

Comment:
Can you add

```js
gtl: {
  loading: null,
},
```

to `miq_global.js` please? That way, you can always rely on it being there. (But more importantly, we get a list of all globals we're using simply by looking at `miq_global.js` ;).)